api/v1beta1/loadbalancer_types.go

```diff
@@ -160,22 +174,51 @@ type LoadBalancerEndpoint struct {
 
 // LoadBalancerInfrastructure defines infrastructure defaults for the LoadBalancer
 type LoadBalancerInfrastructure struct {
+	// Deprecated: use defaultNetwork instead
 	// FloatingNetID defines a openstack ID for the floatingNet.
 	// +optional
 	FloatingNetID *string `json:"floatingNetID,omitempty"`
+	// Deprecated: use defaultNetwork instead
 	// NetworkID defines a openstack ID for the network.
-	NetworkID string `json:"networkID"`
+	// +optional
+	NetworkID string `json:"networkID,omitempty"`
+	// DefaultNetwork defines the default/listener network for the Loadbalancer.
+	// +optional
+	// TODO Remove optional when Deprecations are removed
+	DefaultNetwork LoadBalancerDefaultNetwork `json:"defaultNetwork"`
+	// AdditionalNetworks defines additional networks that will be added to the LoadBalancerMachines.
+	// +optional
+	AdditionalNetworks []LoadBalancerAdditionalNetwork `json:"additionalNetworks"`
 	// Flavor defines openstack flavor for the LoadBalancer. Uses a default if not defined.
 	// +optional
-	Flavor *OpenstackFlavorRef `json:"flavor,omitempty"`
```

Comment:
why? If it's optional shouldn't it be ommited?

Reply:
i think this is because omitempty makes it difficult to set it to `nil` again if it was set before

Reply:
And I think this should not be optional. https://github.com/stackitcloud/yawol/blob/feature/additionalNetworks/controllers/yawol-controller/loadbalancermachine/loadbalancermachine_controller.go#L655
I will fix this.

Reply:
fixed